### PR TITLE
add reserved_key_start_bit option to config hkv

### PR DESCRIPTION
**Pull request by jq**
# Description

Brief Description of the PR:

Fixes # (issue)

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Feature

# Checklist:

- [ ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/recommenders-addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running yapf
    - [ ] By running clang-format
- [ ] This PR addresses an already submitted issue for TensorFlow Recommenders-Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

File: tensorflow_recommenders_addons/dynamic_embedding/core/ops/hkv_hashtable_ops.cc [hunk 1]
```diff
@@ -308,6 +308,7 @@ REGISTER_OP(PREFIX_OP_NAME(HkvHashTableOfTensors))
     .Attr("max_hbm_for_vectors: int = 0")
     .Attr("step_per_epoch: int = 0")
     .Attr("strategy: int = 0")
+    .Attr("reserved_key_start_bit: int = 0")
```
Comment (MoFHeka) [inline tensorflow_recommenders_addons/dynamic_embedding/core/ops/hkv_hashtable_ops.cc:311]: default value of reserved_key_start_bit is better to be same as python code which is 1.

File: tensorflow_recommenders_addons/dynamic_embedding/python/ops/hkv_hashtable_ops.py [hunk 2]
```diff
@@ -79,6 +79,7 @@ def __init__(
       evict_strategy=HkvEvictStrategy.LRU,
       step_per_epoch=0,
       gen_scores_fn=None,
+      reserved_key_start_bit=0,
```
Comment (MoFHeka) [inline tensorflow_recommenders_addons/dynamic_embedding/python/ops/hkv_hashtable_ops.py:82]: A simple explanation of config reserved_key_start_bit in document docs/api_docs/tfra/dynamic_embedding/HkvHashTableConfig.md, that would be better.

**Review by MoFHeka — CHANGES_REQUESTED** [review]
Adding document and a unitive default value of reserved_key_start_bit may be better.

**Review by MoFHeka — APPROVED** [review]
LGTM

**Comment by rhdong** [comment]
Great job!

**Pull request merged**